embulk-cli/src/main/java/org/embulk/cli/EmbulkCommandLine.java

```diff
@@ -0,0 +1,268 @@
+            final List<String> loadPath,
+            final String output,
+            final String resumeState)
+    {
+        this.arguments = Collections.unmodifiableList(arguments);
+        this.systemConfig = Collections.unmodifiableMap(systemConfig);
+        this.bundle = bundle;
+        this.bundlePath = bundlePath;
+        this.classpath = Collections.unmodifiableList(classpath);
+        this.configDiff = configDiff;
+        this.force = force;
+        this.format = format;
+        this.load = Collections.unmodifiableList(load);
+        this.loadPath = Collections.unmodifiableList(loadPath);
+        this.output = output;
+        this.resumeState = resumeState;
+    }
+
+    public static final class Builder
+    {
+        public Builder()
+        {
+            this.arguments = new ArrayList<String>();
+            this.systemConfig = new HashMap<String, Object>();
+            this.bundle = null;
+            this.bundlePath = null;
+            this.classpath = new ArrayList<String>();
+            this.configDiff = null;
+            this.force = false;
+            this.format = null;
+            this.load = new ArrayList<String>();
+            this.loadPath = new ArrayList<String>();
+            this.output = null;
+            this.resumeState = null;
+        }
+
+        public EmbulkCommandLine build()
+        {
+            return new EmbulkCommandLine(
+                this.arguments,
+                this.systemConfig,
+                this.bundle,
+                this.bundlePath,
+                this.classpath,
+                this.configDiff,
+                this.force,
+                this.format,
+                this.load,
+                this.loadPath,
+                this.output,
+                this.resumeState);
+        }
+
+        public Builder addArguments(final List<String> arguments)
+        {
+            this.arguments.addAll(arguments);
+            return this;
+        }
+
+        public Builder setSystemConfig(final String key, final String value)
+        {
+            this.systemConfig.put(key, value);
+            return this;
+        }
+
+        public Builder addSystemConfig(final String key, final String value)
+        {
+            final Object existingValue = this.systemConfig.get(key);
+            if (existingValue != null && existingValue instanceof String) {
+                this.systemConfig.put(key, Arrays.asList((String) existingValue, value));
+            }
+            else if (existingValue != null && existingValue instanceof List) {
+                @SuppressWarnings("unchecked")
+                final ArrayList<String> newList = new ArrayList<String>((List<String>) existingValue);
+                newList.add(value);
+                this.systemConfig.put(key, Collections.unmodifiableList(newList));
+            }
+            else {
+                this.systemConfig.put(key, Arrays.asList(value));
+            }
+            return this;
+        }
```

Comment:
@muga As of the command line (existing `lib/embulk/command/embulk_run.rb`), it processes system configs given from the command line as strings or maps of strings. Even if a user wants to give an integer into `system_config`, the user needs to specify like `-Xsome_config=42`, and that `42` is processed as a string at least in `lib/embulk/command/embulk_run.rb`. (You can grep `system_config` on `lib/embulk/command/embulk_run.rb`.)

All the processes are the same (for the time being) after the new `EmbulkRun.java`. The `system_config`, which is given to Ruby `EmbulkRunner`, is converted into a  pure-Ruby object at: https://github.com/embulk/embulk/pull/543/files/25d5a0273cbe35d01aaef6bf882d021d242b3172#diff-305efcc3bb3a8a54312bb00f55e0c386R453

I wanted `EmbulkRun.java` depending less on other libraries, including Jackson. It should be working well in the same way with the existing `lib/embulk/command/embulk_run.rb`. What do you think?